### PR TITLE
[TEST][REF] Move functions specific to OASIS1 converter in a dedicated utils file and add unit tests

### PR DESCRIPTION
**Pull request by AliceJoubert**
- Move to module specific to OASIS1
- Remove parts specific to OASIS3 (never used)
- Add tests.

### CODE REVIEW

File: test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        tmp_path / "clinical",
+        tmp_path / "BIDS",
+        tmp_path / "spec",
+        ["sub-OASIS10001", "sub-OASIS10002"],
+    )
+    write_sessions_tsv(tmp_path / "BIDS", sessions)
+    sessions_files = list((tmp_path / "BIDS").rglob("*.tsv"))
+    assert len(sessions_files) == 2
+    for file in sessions_files:
+        assert not assert_frame_equal(
```
Comment (NicolasGensollen) [inline test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py:164]: This is weird. What are you trying to assert here ?
I would expect either `assert <boolean condition>` or `assert_frame_equal(df1, df2)`.

File: test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        }
+    )
+    spec.to_csv(tmp_path / "spec" / "sessions.tsv", index=False, sep="\t")
+
+
+@pytest.fixture
+def build_bids_dir(tmp_path: Path) -> None:
+    (tmp_path / "BIDS" / "sub-OASIS10001" / "ses-M000").mkdir(parents=True)
+    (tmp_path / "BIDS" / "sub-OASIS10001" / "ses-M006").mkdir(parents=True)
+    (tmp_path / "BIDS" / "sub-OASIS10002" / "ses-M000").mkdir(parents=True)
```
Comment (NicolasGensollen) [inline test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py:77]: Your code works but is a bit surprising in the way you use fixtures.
Nothing wrong, but I would expect something more or less like this:

```python
@pytest.fixture
def bids_dir(tmp_path: Path) -> Path:
    bids_dir = tmp_path / "BIDS"
    _build_bids_dir(bids_dir)
    return bids_dir

def _build_bids_dir(bids_dir: Path) -> None:
    (bids_dir / "sub...").mkdir(....)
    ...

def test_create_bla_bla_bla(tmp_path, bids_dir: Path, ...):
    result = create_sessions_dict(clinical_data_dir, bids_dir, ...)
    assert result == ...
```

Does this make sense ? WDYT ?

File: test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+
+@pytest.fixture
+def get_expected_dict() -> dict:
```
Comment (NicolasGensollen) [inline test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py:81]: Again, this works fine, but I would simplify by either having a function or a fixture.

That is, either:

```python
def get_expected_dict() -> dict:
    return {...}

def test_bla_bla(tmp_path, ...):
    ...
   assert result == get_expected_dict()
```

Or:

```python
@pytest.fixture
def expected() -> dict:
    return {...}

def test_bla_bla(tmp_path, expected: dict, ...):
    ...
    assert result == expected
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @AliceJoubert !
The code looks good. I made a couple suggestions on the tests in order to improve readability and use fixtures in a more "traditional" way.

File: test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py (hunk 1 above)
Comment (NicolasGensollen) [reply test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py:164]: Maybe something like

```python
assert not df1.equals(df2)
```

**Comment by AliceJoubert** [comment]
Thanks for the review @NicolasGensollen , do you see anything else ?

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @AliceJoubert !

**Pull request merged**